torch/distributed/nn/functional.py

```diff
@@ -219,18 +258,26 @@ def forward(ctx, group, tensor):
 
     @staticmethod
     def backward(ctx, *grad_outputs):
-        gxs = _AlltoAll.apply(ctx.group, *grad_outputs)
+        gxs = _AlltoAll.apply(ctx.group, None, *grad_outputs)
         gx = torch.sum(torch.stack(gxs), dim=0)
         return (None, gx)
 
 
 class _AlltoAll(Function):
     @staticmethod
-    def forward(ctx, group, *tensors):
+    def forward(ctx, group, out_tensor_list, *tensors):
         ctx.group = group
-        out_tensor_list = [
-            torch.empty_like(tensors[i]) for i in range(dist.get_world_size(group=group))
+        ctx.input_tensor_list = [
+            torch.empty_like(tensors[i])
+            for i in range(dist.get_world_size(group=group))
```

Comment:
Since this is only used during backward, why not just save the desired sizes in the ctx and then allocate memory only in the backward method?

Reply:
I see, that makes sense.